### ISSUE TITLE
Full travis builds

### ISSUE THREAD

**Issue by 0xpr03** [fewensa/telegram-client#6]
We're currently not running any real builds on this. Having full builds would allow for direct identification of errors.
I guess this is mainly due to the problematic tdlib dependency. For starters we could host a tdlib.so somewhere (for example git) that travis can download. I could also host a tdlib on my own. If we also enable caching it would reduce build times drastically when rtdlib isn't changed.

**Comment by fewensa** [comment]
Yes , you are right, I already created a repository [tdlib-dylib](https://github.com/fewensa/tdlib-dylib) ,
The function of this repository is to compile td, and upload libtdjson.so to this repository, [build](https://github.com/fewensa/tdlib-dylib/tree/build) branch is compile code, and [master](https://github.com/fewensa/tdlib-dylib/tree/master) branch is save so file.

But I have encountered some problems

[ls: /source/build/libtdjson*: No such file or directory](https://travis-ci.org/fewensa/tdlib-dylib/builds/542612564#L1379)

I don't know why travis  CI can't to copy file , but actually there is, I found some solution, but not work. so [tdlib-dylib](https://github.com/fewensa/tdlib-dylib) was suspended by me. wait unit telegrem-client 
 is stable again.

Maybe you can help me. In my plan, I should compile the dynamic dependency library of all systems. and all version of td [release](https://github.com/tdlib/td/releases) page.

And now i trying to make rtdlib compile faster, expecting good results.

**Comment by tingfeng-key** [comment]
good, i trying build libtdjson, thank

**Comment by dorak88783** [comment]
I could try to help as well - it'd be a challenge but that would be good for me :) 

I'm a bit lost though with all the projects - what's the difference/relation between

https://github.com/fewensa/telegram-client
https://github.com/fewensa/rtdlib-sys
https://github.com/fewensa/tdlib-dylib
https://github.com/fewensa/fantasy

**Comment by dorak88783** [comment]
I made a repository https://github.com/dorak88783/tdlib-builder which has automatic pre-compiled binaries available. You could use those in the CI flow.

**Comment by fewensa** [comment]
Hi @dorak88783  very thanks for you , I think it can be expanded in your repository.

**Comment by fewensa** [comment]
@dorak88783  Your project can support the tdjson.lib of all different versions of different systems? in your plain

**Comment by dorak88783** [comment]
I didn't try yet to build old versions. At least for 1.7 onwards they'll be there, so maybe just add only these tests on the 1.7 and higher versions?
